Source/iOS/OIDAuthorizationUICoordinatorIOS.m

```diff
@@ -71,23 +73,40 @@ - (nullable instancetype)initWithPresentingViewController:
   return self;
 }
 
-- (BOOL)presentAuthorizationWithURL:(NSURL *)URL session:(id<OIDAuthorizationFlowSession>)session {
+- (BOOL)presentAuthorizationRequest:(OIDAuthorizationRequest *)request session:(id<OIDAuthorizationFlowSession>)session {
   if (_authorizationFlowInProgress) {
     // TODO: Handle errors as authorization is already in progress.
     return NO;
   }
 
   _authorizationFlowInProgress = YES;
   _session = session;
-  if ([SFSafariViewController class]) {
-    SFSafariViewController *safariVC =
-        [[[self class] safariViewControllerFactory] safariViewControllerWithURL:URL];
-    safariVC.delegate = self;
-    _safariVC = safariVC;
-    [_presentingViewController presentViewController:safariVC animated:YES completion:nil];
-    return YES;
+  BOOL openedSafari = NO;
+  NSURL *requestURL = [request authorizationRequestURL];
+  
+  if (@available(iOS 11.0, *)) {
+    NSString *redirectScheme = request.redirectURL.scheme;
+    SFAuthenticationSession* authenticationVC =
+        [[SFAuthenticationSession alloc] initWithURL:requestURL
+                                   callbackURLScheme:redirectScheme
+                                   completionHandler:^(NSURL * _Nullable callbackURL,
+                                                       NSError * _Nullable error) {
+      [_session resumeAuthorizationFlowWithURL:callbackURL];
```

Comment:
You probably want to set `_authenticationVC` to `nil` either here or in `cleanUp`.